extensions/shared/i18n-to-php.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+// How (and why) does this transform work?
+// The idea here is to replace every call to each of `@wordpress/i18n`'s translation functions
+// (`__()`, `_n()`, `_x()`, `_nx()`) with their PHP counterpart, wrapped in a `<?php ... ?>`
+// pseudo-tag, and using `echo` to output the var, plus some escaping for sanitization.
+// The most puzzling part might be the `echo`: after all, we can't know _how_ the translated
+// strings are used in the Javascript source -- e.g. they might be assigned to a variable
+// for later usage, rather than rendered directly.
+// The answer is that this happens during React's rendering (to static markup, in this case),
+// where the entire component logic is essentially flattened to a component string. This means
+// that even translated strings that have gone through any intermediate steps will end up in
+// that rendered markup -- and thus, we'll have our `<?php echo esc_html__( ... ) ?>`
+// statements right where they belong.
+// A note on implementation:
+// Ideally, our replaced translation function would simply return `<?php echo esc_html__( ... ) ?>`.
+// However, React sanitizes strings by escaping chars like `<`. As a consequence, we need to use
+// `dangerouslySetInnerHTML` to bypass the escaping. This also requires to be attached as a prop
+// to a DOM element. I've chosen `<span />` since this likely has the smallest footprint for
+// rendering strings (e.g. shouldn't normally get in the way of styling).
+const translate = ( ...args ) => (
+	<span
+		// eslint-disable-next-line react/no-danger
+		dangerouslySetInnerHTML={ {
+			__html: `<?php esc_html_e( ${ args.map( arg => `'${ arg }'` ).join( ', ' ) } ) ?>`,
```

Comment:
Oh, I just noticed, I need to change this a bit, since this always only uses `__()`, but we need the other ones (`_n()`, `_x()`, `_nx()`) in some cases.

Reply:
Added unit tests to establish a contract of what the rendered PHP expressions should looks like, and changed the implementation to fit those.